Automattic/wp-calypso
# Domain management: Transform the MailboxesList component to TSX and create smaller standalone components

Follow up: https://github.com/Automattic/wp-calypso/pull/97252

## Proposed Changes

- Refactor the MailboxesList component, transforming it to a typescript
- Create multiple standalone components for easier maintain

## Why are these changes being made?


* Preparing component for domain management revamp

## Testing Instructions

* Go to `/domains/manage?flags=calypso/all-domain-management`
* Make sure you have your domain (you can purchase a .blog)
* Select the "Email" tab
* Check if the screens are the same as before this PR
* Also, go to `/email/{DOMAIN/manage/{DOMAIN}` (this is current product screen)
* Check if the screens are the same as before this PR

## Pre-merge Checklist
